.travis.yml

```diff
@@ -1,19 +1,41 @@
 language: generic
 dist: xenial
 
-env:
-    - UID=1000 USER=qgis_user
-
 services:
   - docker
 
+cache:
+  directories:
+    - $HOME/docker-cache
+
 before_install:
-  - docker-compose build
+  # If the docker checksum matches, use the cached docker image.
+  - >
+     md5sum --check $HOME/docker-cache/docker-checksum &&
+     cat $HOME/docker-cache/our-image.tar | docker load || true
+
+install:
+  # Rebuild the docker image, but only if the docker checksum does not match.
+  - >
+     md5sum --check $HOME/docker-cache/docker-checksum ||
+     docker-compose build --build-arg uid=`id -u` --build-arg gid=`id -g`
```

Comment:
I feed in uid/gid just like in local development. Reason: travis' user id is 2000, not the default 1000.